src/west/manifest.py

```diff
@@ -89,11 +89,13 @@
 
 # Type for the submodule value passed through the manifest file.
 class Submodule(NamedTuple):
+    '''Represents a Git submodule within a project.'''
+
     name: str
     path: str
 
 # Submodules might be the list of values, bool or None.
```

Comment:
If `SubmodulesType` is not Optional anymore `None` value should be removed from the comment.